src/screencast/pipewire_screencast.c

```diff
@@ -231,6 +231,66 @@ static const struct pw_stream_events pwr_stream_events = {
 	.remove_buffer = pwr_handle_stream_remove_buffer,
 };
 
+void xdpw_pwr_import_buffer(struct xdpw_screencast_instance *cast) {
+	struct pw_buffer *pw_buf;
+
+	logprint(TRACE, "pipewire: importing buffer");
+
+	if ((pw_buf = pw_stream_dequeue_buffer(cast->stream)) == NULL) {
+		logprint(WARN, "pipewire: out of buffers");
+		cast->current_pw_buffer = pw_buf;
+		return;
+	}
+
+	cast->current_pw_buffer = pw_buf;
+}
+
+void xdpw_pwr_export_buffer(struct xdpw_screencast_instance *cast) {
+	struct pw_buffer *pw_buf;
+	struct spa_buffer *spa_buf;
+	struct spa_meta_header *h;
+	struct spa_data *d;
+
+	logprint(TRACE, "pipewire: exporting buffer");
+
+	pw_buf = cast->current_pw_buffer;
+
+	if (!pw_buf) {
```

Comment:
The idea behind that was to ensure, that running the screencopy buffer_done method while the pipewire stream was not established didn't made a problem, but it would probably be better to catch those cases beforehand.

Reply:
rework took care of that and an assert was added.